test/integration/models/test_networking.py

```diff
@@ -15,37 +18,85 @@ def test_get_networking_rules(get_client, create_firewall):
+    chosen_region = available_regions[0]
+    label = get_test_label()
+
+    linode_instance, password = client.linode.instance_create(
+        "g6-nanode-1",
+        chosen_region,
+        image="linode/debian12",
+        label=label,
+    )
+
+    return linode_instance
+
+
+@pytest.fixture
+def create_linode_for_ip_share(get_client):
+    linode = create_linode(get_client)
+
+    yield linode
+
+    linode.delete()
+
+
+@pytest.fixture
+def create_linode_to_be_shared_with_ips(get_client):
+    linode = create_linode(get_client)
+
+    yield linode
+
+    linode.delete()
+
+
 @pytest.mark.smoke
-def test_ip_addresses_share(self):
+def test_ip_addresses_share(
+    get_client, create_linode_for_ip_share, create_linode_to_be_shared_with_ips
+):
     """
     Test that you can share IP addresses with Linode.
     """
-    ip_share_url = "/networking/ips/share"
-    ips = ["127.0.0.1"]
-    linode_id = 12345
-    with self.mock_post(ip_share_url) as m:
-        result = self.client.networking.ip_addresses_share(ips, linode_id)
-
-        self.assertIsNotNone(result)
-        self.assertEqual(m.call_url, ip_share_url)
-        self.assertEqual(
-            m.call_data,
-            {
-                "ips": ips,
-                "linode": linode_id,
-            },
-        )
-
-    # Test that entering an empty IP array is allowed.
-    with self.mock_post(ip_share_url) as m:
-        result = self.client.networking.ip_addresses_share([], linode_id)
-
-        self.assertIsNotNone(result)
-        self.assertEqual(m.call_url, ip_share_url)
-        self.assertEqual(
-            m.call_data,
-            {
-                "ips": [],
-                "linode": linode_id,
-            },
-        )
+
+    # create two linode instances and share the ip of instance1 with instance2
+    linode_instance1 = create_linode_for_ip_share
+    linode_instance2 = create_linode_to_be_shared_with_ips
+
+    get_client.networking.ip_addresses_share(
+        [linode_instance1.ips.ipv4.public[0]], linode_instance2.id
+    )
+
+    assert (
+        linode_instance1.ips.ipv4.public[0].address
+        == linode_instance2.ips.ipv4.shared[0].address
+    )
+
+
+@pytest.mark.smoke
+def test_ip_addresses_unshare(
+    get_client, create_linode_for_ip_share, create_linode_to_be_shared_with_ips
+):
+    """
+    Test that you can unshare IP addresses with Linode.
+    """
+
+    # create two linode instances and share the ip of instance1 with instance2
+    linode_instance1 = create_linode_for_ip_share
+    linode_instance2 = create_linode_to_be_shared_with_ips
+
+    get_client.networking.ip_addresses_share(
+        [linode_instance1.ips.ipv4.public[0]], linode_instance2.id
+    )
+
+    # unshared the ip with instance2
+    get_client.networking.ip_addresses_share([], linode_instance2.id)
+
+    assert [] == linode_instance2.ips.ipv4.shared
+
+
+def get_test_label():
+    unique_timestamp = str(int(time.time()) + random.randint(0, 1000))
```

Comment:
```suggestion
    unique_timestamp = str(time.time_ns())
```

Reply:
Yeah I agree with this.  I have used the time with random int in the past but, I think `time.time_ns()` is much cleaner solution.
`str(int(time.time_ns()))` int() will remove all the decimal values